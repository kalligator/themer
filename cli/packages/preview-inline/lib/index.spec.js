const { render } = require('./index');
const { colors } = require('../../colors-default');

describe('themer color set inline preview', () => {
  it('should use the color set name as the default name', (done) => {
    Promise.all(render(colors, { colors: 'colors-default' })).then((files) => {
      expect(
        files.every((file) =>
          /colors-default-(light|dark)-inline\.svg/.test(file.name),
        ),
      ).toBe(true);
      done();
    });
  });
  it('should also accept custom names through options', (done) => {
    Promise.all(
      render(colors, {
        'themer-preview-inline-name': 'test',
        'colors': 'colors-default',
      }),
    ).then((files) => {
      expect(
        files.every((file) => /test-(light|dark)-inline\.svg/.test(file.name)),
      ).toBe(true);
      done();
    });
  });
  it('should render proper SVG files', (done) => {
    Promise.all(render(colors, { colors: 'colors-default' })).then((files) => {
      files.forEach((file) => {
        expect(file.contents.toString('utf8')).toMatchSnapshot();
      });
      done();
    });
  });
});
