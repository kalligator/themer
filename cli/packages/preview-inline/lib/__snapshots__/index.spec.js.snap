// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer color set inline preview should render proper SVG files 1`] = `
"
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 160 24\\">

          <rect fill=\\"#282629\\" x=\\"0\\" y=\\"0\\" width=\\"160\\" height=\\"24\\"></rect>

          <rect fill=\\"#474247\\" x=\\"8\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#656066\\" x=\\"29\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#847E85\\" x=\\"50\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#A29DA3\\" x=\\"71\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#C1BCC2\\" x=\\"92\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#E0DCE0\\" x=\\"113\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#FFFCFF\\" x=\\"134\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>

          <rect fill=\\"#FF4050\\" x=\\"8\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#F28144\\" x=\\"27\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#FFD24A\\" x=\\"46\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#A4CC35\\" x=\\"65\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#26C99E\\" x=\\"84\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#66BFFF\\" x=\\"103\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#CC78FA\\" x=\\"122\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#F553BF\\" x=\\"141\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>

        </svg>
      "
`;

exports[`themer color set inline preview should render proper SVG files 2`] = `
"
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 160 24\\">

          <rect fill=\\"#FFFCFF\\" x=\\"0\\" y=\\"0\\" width=\\"160\\" height=\\"24\\"></rect>

          <rect fill=\\"#E0DCE0\\" x=\\"8\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#C1BCC2\\" x=\\"29\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#A29DA3\\" x=\\"50\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#847E85\\" x=\\"71\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#656066\\" x=\\"92\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#474247\\" x=\\"113\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#282629\\" x=\\"134\\" y=\\"13\\" width=\\"18\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>

          <rect fill=\\"#F03E4D\\" x=\\"8\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#F37735\\" x=\\"27\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#EEBA21\\" x=\\"46\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#97BD2D\\" x=\\"65\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#1FC598\\" x=\\"84\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#53A6E1\\" x=\\"103\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#BF65F0\\" x=\\"122\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>
          <rect fill=\\"#EE4EB8\\" x=\\"141\\" y=\\"8\\" width=\\"11\\" height=\\"3\\" rx=\\"1\\" ry=\\"1\\"></rect>

        </svg>
      "
`;
